src/runtime/virtcontainers/kata_agent.go

```diff
@@ -1250,12 +1250,17 @@ func (k *kataAgent) createContainer(ctx context.Context, sandbox *Sandbox, c *Co
 		return nil, err
 	}
 
-	if sharedRootfs.storage != nil {
-		// Add rootfs to the list of container storage.
-		// We only need to do this for block based rootfs, as we
+	if sharedRootfs.volumeStorages != nil {
+		// Add volumeStorages to the list of container storage.
+		// We only need to do this for KataVirtualVolume based rootfs, as we
 		// want the agent to mount it into the right location
-		// (kataGuestSharedDir/ctrID/
-		ctrStorages = append(ctrStorages, sharedRootfs.storage)
+
+		ctrStorages = append(sharedRootfs.volumeStorages, ctrStorages...)
+	}
+
+	if sharedRootfs.containerStorages != nil {
```

Comment:
Should we add `containerStorages` before `volumeStorages`? Logically volume storage depends on container storage(rootfs).

Reply:
Fixed.